features/mbedtls/targets/TARGET_NUVOTON/TARGET_NUC472/sha/sha_alt_hw.c

```diff
@@ -102,17 +94,16 @@ void mbedtls_sha1_hw_finish(crypto_sha_context *ctx, unsigned char output[20])
         crypto_sha_update_nobuf(ctx, ctx->buffer, ctx->buffer_left, 1);
         ctx->buffer_left = 0;
         crypto_sha_getinternstate(output, 20);
```

Comment:
@mazimkhan No. In C/C++, array argument is treated as pointer argument. In this case, `sizeof(output)` is equal to `sizeof(unsigned char *)` rather than expected `sizeof(unsigned char) * 20`. Besides, because SHA output size 20 is not defined in `mbedtls/sha1.h`, I follow it by just using constant.

Reply:
Ok. I have never used array arguments myself. 
But then what is the point of having array arguments? They don't even guaranty argument type & size and work more like a red herring for unaware user. I tried it with following code. It only warns about the ```sizeof````:
```
#include <stdio.h>

void foo(const char o[16])
{
    printf ("sizeof(o) = %lu\n", sizeof(o));
}

int main()
{
    char o[64];
    char * str = NULL;
    foo(o);
    foo(str);
    return 0;
}
```
Compilation output:
```
mazimkhan@mazimkhan-VirtualBox:~$ g++ main.c 
main.c: In function ‘void foo(const char*)’:
main.c:5:42: warning: ‘sizeof’ on array function parameter ‘o’ will return size of ‘const char*’ [-Wsizeof-array-argument]
     printf ("sizeof(o) = %lu\n", sizeof(o));
                                          ^
main.c:3:25: note: declared here
 void foo(const char o[16])
                         ^
mazimkhan@mazimkhan-VirtualBox:~$ ./a.out 
sizeof(o) = 8
sizeof(o) = 8
```

Reply:
I would not ask to change above code since Mbed TLS has lot of such examples. I will take a look into it to see if there are any merits in having array arguments.